data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactory.java

```diff
@@ -288,10 +295,22 @@ private SSLContext getCAStrategy(final ConnectionConfiguration connectionConfigu
         final Path certPath = connectionConfiguration.getCertPath();
         if (Objects.nonNull(certPath)) {
             return TrustStoreProvider.createSSLContext(certPath);
-        } else if (Objects.nonNull(connectionConfiguration.getCertificateContent())) {
-            return TrustStoreProvider.createSSLContext(connectionConfiguration.getCertificateContent());
+        } else if (Objects.nonNull(connectionConfiguration.getCertificate())) {
+            if (isPemObject(connectionConfiguration.getCertificate())) {
+                return TrustStoreProvider.createSSLContext(connectionConfiguration.getCertificate());
+            } else {
+                return TrustStoreProvider.createSSLContext(Path.of(connectionConfiguration.getCertificate()));
+            }
         } else {
             return TrustStoreProvider.createSSLContextWithTrustAllStrategy();
         }
     }
+
+    private boolean isPemObject(final String certificate) {
+        try (PemReader reader = new PemReader(new StringReader(certificate))) {
+            return reader.readPemObject() != null;
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to read certificate.", e);
```

Comment:
Let's have this return a `false`. We can rely on the final `else` in the caller to determine what to do next.